06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js

```diff
@@ -0,0 +1,186 @@
+    return token;
+  }
+  /**
+   * Reads the data from an uploaded file.
+   * @param {File} file - The file to be read
+   * @param {Function} callback - The callback function on file read
+   * @returns {void}
+   */
+  readFile(file, callback) {
+    const reader = new FileReader();
+    reader.onload = callback;
+    reader.readAsText(file);
+  }
+/**
+   * It validates the file passed and returns the error message if an error occurs
+   *
+   * @method validateFile
+   * @param {String} file
+   *
+   * @return {object} validates the file passed
+   */
+  validateFile(file) {
+    const jsonFile = file;
+    let check = {
+      status: true,
+      msg: 'Valid File',
+    };
+    try {
+      if (typeof file !== 'object' || file.length < 1) {
+        check = {
+          status: false,
+          msg: 'File is empty please upload a new file',
+        };
+      }
+      jsonFile.forEach((key) => {
+        if (key.title === undefined || key.text === undefined) {
+          check = {
+            status: false,
+            msg: 'Invalid file content',
+          };
+        }
+      });
+    } catch (error) {
+      return error.msg;
+    }
+  }
+/**
+     * It returns the index of the file passed
+     * @method createIndex
+     * @param {Object, Object} fileName jsonObject
+     * @return {number}
+     */
+  createIndex(fileName, jsonObject) {
+    const newIndex = {};
+    this.validateFile(jsonObject);
+
+    jsonObject.forEach((object, position) => {
+      const longSentence = `${object.title} ${object.text}`;
+      const tokenized = this.tokenize(longSentence);
+      const wordArray = tokenized.split(' ');
+
+      wordArray.forEach((word, pos) => {
+        if (newIndex[word] === undefined) {
+          newIndex[word] = [position];
+        } else if (newIndex[word].indexOf(position) < 0) {
+          newIndex[word].push(position);
+        }
+      });
+    });
+    this.index[fileName] = newIndex;
+  }
+/**
+     * It returns the index of the words
+     * @method getIndex
+     * @param {Object} fileName
+     *
+     * @return {object} gets the index
+     */
+  getIndex(fileName) {
+    if (Object.keys(arguments).length < 1) {
+      return this.index;
+    }
+
+    return this.index[fileName];
+  }
+  /**
+   *
+   *
+   * @method flattenSearch
+   * @param {void}
+   *
+   * @return {void}
+   */
+  flattenSearch() {
+    this.temp_search = [];
+    for (const arg of arguments) {
+      if (arg instanceof Object && typeof arg !== 'string') {
+        for (const item in arg) {
+          if (arg.hasOwnProperty(item)) {
+            this.flattenSearch(arg[item]);
+          }
+        }
+      } else {
+        const args = arg.split(' ');
+        args.forEach((word) => {
+          this.temp_search.push(word);
+        });
+      }
+    }
+  }
+/**
+     * it return the word searched for in the object it was found
+     *
+     * @method readFile
+     * @param {String} sTerms
+     * @param {String} allIndex
```

Comment:
Are you using 'allIndex' anywhere in this method? Eslint flagged this error here - _Expected JSDoc for 'fileName' but found 'allIndex'._